### ISSUE TITLE
Testcafe changes AJAX Requests

### ISSUE THREAD

**Issue by MBurchard** [DevExpress/testcafe#3634]
### What is your Test Scenario?
We would like to test a Wicket application that frequently uses AJAX queries.
Large parts of the page are changed by AJAX Behaviour.
 
### What is the Current behavior?
The AJAX request is answered by the server, but Testcafe builds its own HTML code for it...

Because of the manipulated Response the Application does not work.

Original HTML

    <?xml version="1.0" encoding="UTF-8"?><ajax-response><component id="paneld6d" ><![CDATA[<div class="tab-panel" id="paneld6d" data-wicketSource="de.bayern.stmelf.grundstuecksverkehrsgesetz.web.kvb:VorgangErfassenPage.java:283">
        <h2>Veräußerer erfassen</h2>
        ...

Manipulated HTML

    <html><head><meta class="charset-hammerhead-shadow-ui" charset="utf-8"><link rel="stylesheet" type="text/css" class="ui-stylesheet-hammerhead-shadow-ui" href="http://10.36.25.128:51248/testcafe-ui-styles.css"><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://10.36.25.128:51248/hammerhead.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://10.36.25.128:51248/testcafe-core.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://10.36.25.128:51248/testcafe-ui.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://10.36.25.128:51248/testcafe-automation.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://10.36.25.128:51248/testcafe-driver.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://10.36.25.128:51248/iframe-task.js"></script></head><body><script class="self-removing-script-hammerhead-shadow-ui">(function () {if (window["%hammerhead%"])window["%hammerhead%"].sandbox.node.raiseBodyCreatedEvent();var currentScript = document.currentScript;var scriptsLength;var scripts;if (!currentScript) {var hammerhead;try {hammerhead = parent["%hammerhead%"] || window["%hammerhead%"];}catch (e) {hammerhead = window["%hammerhead%"];}if (hammerhead) {try {scripts       = hammerhead.nativeMethods.documentScriptsGetter.call(document);scriptsLength = hammerhead.nativeMethods.htmlCollectionLengthGetter.call(scripts);}catch (e) {}}scripts       = scripts || document.scripts;scriptsLength = scriptsLength !== void 0 ? scriptsLength : scripts.length;currentScript = scripts[scriptsLength - 1];}currentScript.parentNode.removeChild(currentScript);})();</script><ajax-response><component id="panel1a0">
        <h2>Veräußerer erfassen</h2>
        ...

### What is the Expected behavior?
Since we have not set up a HOOK, Testcafe should not manipulate this request.
 
### What is your web application and your TestCafe test code?
This is not possible.
 
Your website URL (or attach your complete example): Internal

Test Call:
```js
  await t.click(Selector('input').withAttribute('data-ref', 'addDetails'));
```
### Your Environment details:
 
* testcafe version:                   1.1.0
* node.js version:                    v10.15.1
* command-line arguments:  testcafe firefox testVerwaltung.js
* command-line arguments:  testcafe ie testVerwaltung.js
* browser name and version: IE 11, Firefox 60.3.0esr
* platform and version:          Windows 10 1809
* Apache Wicket:                   7.10.0

**Comment by LavrovArtem** [comment]
This is definitely incorrect that this information is added to the ajax response on your side.  However, I was not able to reproduce this behavior using Wicket examples (e.g. http://examples8x.wicket.apache.org/ajax/autocomplete).  Thus, the information we currently have is not sufficient to find the cause of the issue. Could you please provide a small example so that we can reproduce it on our side?

**Comment by MBurchard** [comment]
Yes I will prepare a test case that can be downloaded as ZIP. Please give me some time.

**Comment by LavrovArtem** [comment]
Thanks, we'll wait for it.

**Comment by MBurchard** [comment]
I prepared a test case: https://github.com/MBurchard/testcafe-3634

But don't rejoice too soon, the test case doesn't lead to the problem.

That's why I took two screenshots. Both are take on the same machine with same version of Testcafe.

![image001](https://user-images.githubusercontent.com/4256302/55570656-1c6e7f00-5704-11e9-8ba4-2671bb0df0d3.png)

The first one is from the testcase I provided. This simply starts a small SpringBoot App locally.

![image002](https://user-images.githubusercontent.com/4256302/55570681-27c1aa80-5704-11e9-9cf3-75c12ddf480c.png)

The second one starts a complex portal framework on the local tomcat.

As one can see, the headers are a little different.

May it be, that because of Content-Type or other header parameters Testcafe injects some content into the AJAX Response?

Is this hint sufficient for further research?

What else could I do?
